src/torchvision_module.py

```diff
@@ -0,0 +1,291 @@
+            raise NotImplementedError
+        input_tensor = self.preprocessor(image)
+        with torch.no_grad():
+            prediction: Tensor = self.model(input_tensor)[0]
+        labels = [self.weights.meta["categories"][i] for i in prediction["labels"]]
+        scores = prediction["scores"]
+        boxes = prediction["boxes"].to(torch.int64).tolist()
+        res = [
+            Detection(
+                x_min=x_min,
+                y_min=y_min,
+                x_max=x_max,
+                y_max=y_max,
+                confidence=score,
+                class_name=label,
+            )
+            for (x_min, y_min, x_max, y_max), score, label in zip(boxes, scores, labels)
+        ]
+        res = self.filter_output(res)
+        return res
+
+    async def get_classifications(
+        self,
+        image: Union[Image.Image, RawImage],
+        count: int,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ) -> List[Classification]:
+        if not self.properties.implements_classification:
+            return NotImplementedError
+
+        input_tensor = self.preprocessor(image)
+        # input_tensor = image
+        # torchvision.utils.save_image(input_tensor, "./input.jpg")
+        # img = decode_image(image)
+        with torch.no_grad():
+            prediction: Tensor = self.model(input_tensor)
+
+        prediction = prediction.squeeze(0)
+        prediction = prediction.softmax(0)
+        scores, top_indices = torch.topk(prediction, k=count)
+
+        category_names = [
+            self.weights.meta["categories"][index] for index in top_indices
+        ]
+        res = [
+            Classification(class_name=name, confidence=score)
+            for name, score in zip(category_names, scores)
+        ]
+        res = self.filter_output(res)
+        return res
+
+    async def get_classifications_from_camera(
+        self,
+        camera_name: str,
+        count: int,
+        *,
+        extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None,
+    ) -> List[Classification]:
+        if not self.properties.implements_classification:
+            raise NotImplementedError
+
+        image = await self.get_image_from_dependency(camera_name)
+        input_tensor = self.preprocessor(image)
+        with torch.no_grad():
+            prediction: Tensor = self.model(input_tensor)
+
+        prediction = prediction.squeeze(0)
+        prediction = prediction.softmax(0)
+        scores, top_indices = torch.topk(prediction, k=count)
+
+        category_names = [
+            self.weights.meta["categories"][index] for index in top_indices
+        ]
+        res = [
+            Classification(class_name=name, confidence=score)
+            for name, score in zip(category_names, scores)
+        ]
+        res = self.filter_output(res)
+        return res
+
+    async def get_detections_from_camera(
```

Comment:
same with here - don't repeat code, but put the shared code in a third function. In that way, if you ever have to edit this code, you can avoid a bug where you change it in one function, but not another

Reply:
done